proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,670 @@
+```
+LIP: <LIP number>
+Title: Introduce cross-chain messages
+Author: Maxime Gagnebin <maxime.gagnebin@lightcurve.io>
+Discussions-To: https://research.lisk.com/t/introduce-cross-chain-messages/299
+Type: Standards Track
+Created: <YYYY-MM-DD>
+Updated: <YYYY-MM-DD>
+Requires: Introduce cross-chain update mechanism
+```
+
+
+## Abstract
+
+This proposal introduces the cross-chain message schema, the generic message processing and the base error handling. 
+Defining a base cross-chain message allows all chains in the ecosystem to read and understand the base properties of messages.
+
+The proposal also introduces four messages used by the Interoperability module, the cross-chain update receipt, the channel terminated message, the sidechain terminated message and the registration message.
+
+
+## Copyright
+
+This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
+
+
+## Motivation
+
+To achieve interoperability, chains need to exchange information by sending messages to each other. 
+To this end, we introduce a new message format, which we call a _cross-chain message_ and define in this proposal the base schema of cross-chain messages. 
+Specifying a unified base message schema allows all chains in the Lisk ecosystem to deserialize and read cross-chain messages. 
+
+To achieve some of its basic functionalities, the Interoperability module uses four messages. 
+The first one, the _cross-chain update receipt_, is sent back to the partner chain (the chain with which the current chain is exchanging messages) whenever a cross-chain update transaction from this chain gets included. 
+The cross-chain update receipt serves to update a few properties of the interoperability store, and in general to acknowledge the inclusion of cross-chain update transactions. 
+The second one, the _channel terminated message_, is sent to chains which have been terminated. 
+This message also updates a few properties of the interoperability store to keep them synchronized between chains. 
+The third one, the _registration message_, is used when registering a chain. 
+This message activates the channel between the chains.
+Furthermore, it serves as a guarantee that the correct chain ID and network ID were used when the registration transaction was sent on the sidechain.
```

Comment:
```suggestion
Furthermore, it serves as a guarantee that the correct name and network ID were used when the registration transaction was sent on the sidechain.
```

Reply:
> registration transaction was sent on the sidechain.
Shouldn't it also be "was sent on the other chain."